Compatibility with Firefox 36-43

JeanPaulLucien opened the pull request:
- replaced 'let map;' by 'var map;'
- reworked map.js to make compatible
- add script to detect browser version
- add cases for ECMA 5 and 6 to map.html

Tested in 45.9 ESR and 58

MichaelKohler commented on js/detectBrowserVersion.js:
```diff
@@ -0,0 +1,22 @@
+	    ecmaVersion = 6,
+	    userAgent = navigator.userAgent;
+
+if (navigator.userAgent.search(/Firefox/) > -1) {
```
as this only works for Firefox, I'm really hesitant to accept this PR. See the other comment for my suggestion.

MichaelKohler commented on js/getCountryCode.js:
```diff
@@ -0,0 +1,258 @@
+function getCountryCode(countryName) {
```
before this change the getCountryCode function was in a closure and therefore not globally accessible, now it is. I always try to not clutter the global namespace.

JeanPaulLucien replied:
The getCountryCode is data strings. I think to store a different entities apart is good idea. Global still works inside the same tab. Right? I know that copy with same name may break script if somebody try to change this somewhere else. Do you think there is high chance for this case or others issues?

Maybe code could be better.

MichaelKohler replied:
Sure, it would work. What we could do here is using our own namespace so the risk of having another function overwriting this one is very small. Something like

```
var FirefoxSprintCountries = {
  getCountryCody: function(countryName) {
    ..
  }
};
```

and then calling it with

```
FirefoxSprintCountries.getCountryCode('somecode');
```

What do you think?

JeanPaulLucien commented:
Michael, thank you for review.

There is ready a commit. I have tried to allow the remarks. This still works on 42, 45 ESR, 58 Nightly:
- rewrote const to vars in mapData.js
- replaced getStyles(); to map.getStyles.js
- created namespaces ffQSprint for getCountryCode and getStyles
- deleted alert(); from map.html about out-of-date browser
- fixed switch in map.html, if something will go wrong
- readability improvement for mapECMA5.js, but it is still Babel

Do I need to send a pull request to verify?

MichaelKohler commented:
As I said before, we don't need an explicit file with ES5, just the normal file we already have rewritten in readable ES5 is fine.

If you want me to take a look and do a review, I'm happy to do so in a new PR.
